core/app/models/spree/taxonomy.rb

```diff
@@ -2,10 +2,11 @@ module Spree
   class Taxonomy < Spree::Base
     acts_as_list
 
-    validates :name, presence: true, uniqueness: { case_sensitive: false, allow_blank: true }
+    validates :name, presence: true, uniqueness: { case_sensitive: false, allow_blank: true, scope: :store_id }
 
     has_many :taxons, inverse_of: :taxonomy
     has_one :root, -> { where parent_id: nil }, class_name: 'Spree::Taxon', dependent: :destroy
+    belongs_to :store, class_name: 'Spree::Store', optional: false
```

Comment:
@szymoniwacz this will not work in spree, we need to add `validates :store, presence: true` - please see: https://github.com/spree/spree/blob/master/core/app/models/spree/base.rb#L21